project_one/lib/project_one.ex

```diff
@@ -0,0 +1,33 @@
+defmodule ProjectOne do
+  @moduledoc """
+  Documentation for `ProjectOne`.
+  """
+
+  @doc """
+  Hello chmod.
+
+  ## Examples
+
+      iex> ProjectOne.calculate_chmod(%{user: "rwx", group: "r-x", other: "r-x"})
+      "755"
+
+  """
+  def calculate_chmod(perm) do
+    user = Map.get(perm, :user)
+    group = Map.get(perm, :group)
+    other = Map.get(perm, :other)
+    "#{calculate_permission(user)}#{calculate_permission(group)}#{calculate_permission(other)}"
+  end
+
+  defp calculate_permission(nil), do: "0"
+
+  defp calculate_permission(code) do
+    {num, _} =
+      code
+      |> String.replace(~r/\w/, "1")
+      |> String.replace(~r/-/, "0")
+      |> Integer.parse(2)
+
+    num
+  end
```

Comment:
Мне очень нравится решение с распознаванием двоичного числа, но мне пришлось потратить секунд пять, чтобы понять, как оно работает. Не нужно загадывать загадки своим кодом, если можно проще :)

```suggestion
  defp calculate_permission("rwx"), do: 7
  defp calculate_permission("rwx"), do: 7
  defp calculate_permission("rw-"), do: 6
  defp calculate_permission("r-x"), do: 5
  defp calculate_permission("r--"), do: 4
  defp calculate_permission("-wx"), do: 3
  defp calculate_permission("-w-"), do: 2
  defp calculate_permission("--x"), do: 1
  defp calculate_permission("---"), do: 0
```

Теперь `"gbp"` выбросит исключение, что правильно.

Reply:
Спасибо, свои косяки осознал и раскаялся.
Позволю себе сделать ещё один заход в сторону решения через распознание двоичных чисел. Фиксы добавлены.